src/components/FilterBar/FilterBar.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import style from './FilterBar.module.css';
+import TuneIcon from '@mui/icons-material/Tune';
+import GridViewRoundedIcon from '@mui/icons-material/GridViewRounded';
+import ViewDayOutlinedIcon from '@mui/icons-material/ViewDayOutlined';
+import BasicSelect from '../BasicSelect/BasicSelect';
+import { TextField } from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+import useStyles from './filterBarStyle';
+import Menu from '@mui/material/Menu';
+import FilterMenu from '../FilterMenu/FilterMenu';
+import { useState,useContext} from 'react';
+import {dataContext,viewContext} from '../ShoppingProducts/ShoppingProducts';
+import { Box } from '@mui/material';
+import Typography from '@mui/material/Typography';
+
+const FilterBar = () => {
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const open = Boolean(anchorEl);
+  const [data, setData] = useContext(dataContext);
+  const dataLimited=[...data];
+  const [limit,setLimit]=useState(30);
+  const [isViewSolo,setIsViewSolo]=useContext(viewContext);
+  
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleChange = (event) => {
+    setLimit(event.target.value);
+  };
+const handleLimiting=(event)=>{
+  // if(event.target.value<data.length)
+  setData(dataLimited.slice(0, event.target.value));
+  console.log(dataLimited.slice(0, event.target.value));
+}
+
+const handleToggleToSolo = () => {
```

Comment:
remove one of those two functions , and make one like this:

const handleToggle = ( value ) => {
   setIsViewSolo(value);
};